R/support.R

```diff
@@ -2,13 +2,14 @@
 #'
 #' @param x A list of prototype vectors defining the distribution type.
 #' @param limits A list of value limits for the distribution.
+#' @param interval A list of interval types for the distribution.
 #'
-new_support_region <- function(x, limits = NULL) {
-  vctrs::new_rcrd(list(x = x, lim = limits), class = "support_region")
+new_support_region <- function(x, limits = NULL, interval = list(c('closed','closed'))) {
```

Comment:
we can do a logical vector, but it still needs to be logical(2L) because the ends of the interval can be different (open-open, open-closed, closed-open, closed-closed).